nirps/qc_checks/check_coordinates.py

```diff
@@ -0,0 +1,137 @@
+
+    # Set observer's location
+    observer_location = EarthLocation(observer_longitude*uu.deg, observer_latitude*uu.deg)
+
+    # Set observation time
+    observation_time = Time(observation_time)
+
+    # Transform coordinates to Altitude and Azimuth
+    altaz_coordinates = coordinates.transform_to(AltAz(obstime=observation_time, location=observer_location))
+
+    # Return Altitude and Azimuth
+    return altaz_coordinates.alt.deg*uu.deg, altaz_coordinates.az.deg*uu.deg
+
+def coords_check(fits_file, RA, DEC, tol=1/60, verbose=False): 
+    # Read in the fits file
+    obj_data = fits.open(fits_file, hdu=1)
+    header = obj_data[0].header
+
+    # Fetch observation infos
+    ALT, AZ = header['HIERARCH ESO TEL ALT']*uu.deg, header['HIERARCH ESO TEL AZ']*uu.deg
+    date_obs = header['DATE-OBS']
+    observer_latitude, observer_longitude = header['HIERARCH ESO TEL GEOLAT'], header['HIERARCH ESO TEL GEOLON']
+    
+    # Predict Altitude and Azimuth
+    pred_ALT, pred_AZ = ra_dec_to_alt_az(RA, DEC, observer_latitude, observer_longitude, date_obs)
+    pred_AZ = pred_AZ - 180*uu.deg # Southern hemisphere
+    
+    # Compute the error in position
+    sin_error_ALT = np.abs(np.sin(ALT) - np.sin(pred_ALT)) # sin(x) = x for small x
+    sin_error_AZ = np.abs(np.sin(AZ) - np.sin(pred_AZ))
+    
+    if verbose: print('Position error: delta ALT = {} deg, delta AZ = {} deg'.format(sin_error_ALT, sin_error_AZ))
+    
+    if sin_error_ALT > tol or sin_error_AZ > tol:
+        if verbose: print('Position error is too large!')
+        return False
+    else: 
+        if verbose: print('Position error is within tolerance.')
+        return True
+    
+
+# ## Fetching the target's info 
+
+obj_name = 'LHS1140'
+
+MAIN_URL = ('https://docs.google.com/spreadsheets/d/'
+            '1dOogfEwC7wAagjVFdouB1Y1JdF9Eva4uDW6CTZ8x2FM/'
+            'export?format=csv&gid=0')
+PENDING_URL = ('https://docs.google.com/spreadsheets/d/'
+               '1dOogfEwC7wAagjVFdouB1Y1JdF9Eva4uDW6CTZ8x2FM/'
+               'export?format=csv&gid=623506317')
+
+# get main table
+main_request = requests.get(MAIN_URL)
+# open main table
+main_dataframe = pd.read_csv(StringIO(main_request.text))
+# get pending table
+pending_request = requests.get(PENDING_URL)
+# open pending table
+pending_dataframe = pd.read_csv(StringIO(pending_request.text))
+# merge these keeping all rows from main table and adding non-repeating
+#  rows from pending table
+astrom_dataframe = pd.concat([main_dataframe,
+                                pending_dataframe]).drop_duplicates()
+
+# if you prefer an astropy table
+astrom_table = Table.from_pandas(astrom_dataframe)
+
+itarget = np.where(astrom_table['OBJNAME'] == obj_name)[0][0]
+target_table = astrom_table[itarget]
+
+
+# Propogate the star w.r.t its proper motion
+
+def propagate_coords(objdata: Dict[str, Any], obs_time: Time) -> Tuple[SkyCoord, Time]:
+    # deal with distance
```

Comment:
Try not to mix functions and "main" code.

1. Description of code
2. Imports
3. Global variables
4. functions
5. main code (`if __name__ == '__main__'`)